pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartitionUpsertMetadataManager.java

```diff
@@ -64,6 +68,14 @@ public interface PartitionUpsertMetadataManager extends Closeable {
    */
   void addSegment(ImmutableSegment segment);
 
+  /**
+   * Preload segments for the table partition. Segments can be added differently during preloading.
+   */
+  void preloadSegments(TableDataManager tableDataManager, IndexLoadingConfig indexLoadingConfig,
```

Comment:
Suggest only passing in `indexLoadingConfig`. Other properties can be set during initialization. We can consider only setting `tableDataManager` during initialization, and add getters in `TableDataManager` to get `helixManager` and `segmentPreloadExecutor`